File: keyansys/src/com/keyansys/hibernate/beans/Paper.java
package com.keyansys.hibernate.beans;

import java.sql.Timestamp;
import java.util.Date;

/**
 * Paper entity. @author MyEclipse Persistence Tools
 */

public class Paper implements java.io.Serializable {

	// Fields

	private Integer paper_id;
	private String name;
	private String username;
	private String department;
	private String paper_name;
	private String first_author;
	private String periodicalName;
	private String issn;
	private String cn;
	private String volume;
	private String core;
	private String pursuant;
	private String approval;
	
	private Date publish_time;
	private Date last_update;

	// Constructors

	/** default constructor */
	public Paper() {
	}

	/** minimal constructor */
	public Paper(Date publish_time) {
		this.publish_time = publish_time;
	}
	
	public Paper(String name,String username, String department, String first_author, String paper_name, String pursuant ) 
	{
		this.name = name;
		this.username = username;
		this.department = department;
		this.first_author = first_author;
		this.paper_name = paper_name;
		this.pursuant = pursuant;
		
		
	}

	/** full constructor */
	public Paper(Integer paper_id,String name,String username, String department, String paper_name,
			String first_author, String periodicalName, String issn, String cn,
			String volume, String core, String approval, String workload,
			Date publishTime, Date last_update) {
		this.paper_id=paper_id;
		this.name = name;
		this.username = username;
		this.department = department;
		this.paper_name = paper_name;
		this.first_author = first_author;
		this.periodicalName = periodicalName;
		this.issn = issn;
		this.cn = cn;
		this.volume = volume;
		this.core = core;
		this.approval = approval;
		
		this.publish_time = publishTime;
		this.last_update = last_update;
	}

	// Property accessors

	public Integer getPaper_id() {
		return this.paper_id;
	}

	public void setPaper_id(Integer paper_id) {
		this.paper_id = paper_id;
	}

	public String getUsername() {
		return this.username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getDepartment() {
		return this.department;
	}

	public void setDepartment(String department) {
		this.department = department;
	}

	public String getPaper_name() {
		return this.paper_name;
	}

	public void setPaper_name(String paper_name) {
		this.paper_name = paper_name;
	}

	public String getFirst_author() {
		return this.first_author;
	}

	public void setFirst_author(String first_author) {
		this.first_author = first_author;
	}

	public String getPeriodicalName() {
		return this.periodicalName;
	}

	public void setPeriodicalName(String periodicalName) {
		this.periodicalName = periodicalName;
	}

	public String getIssn() {
		return this.issn;
	}

	public void setIssn(String issn) {
		this.issn = issn;
	}

	public String getCn() {
		return this.cn;
	}

	public void setCn(String cn) {
		this.cn = cn;
	}

	public String getVolume() {
		return this.volume;
	}

	public void setVolume(String volume) {
		this.volume = volume;
	}

	public String getCore() {
		return this.core;
	}

	public void setCore(String core) {
		this.core = core;
	}

	public String getPursuant() {
		return pursuant;
	}

	public void setPursuant(String pursuant) {
		this.pursuant = pursuant;
	}

	public String getApproval() {
		return this.approval;
	}

	public void setApproval(String approval) {
		this.approval = approval;
	}

	

	public Date getPublish_time() {
		return this.publish_time;
	}

	public void setPublish_time(Date publish_time) {
		this.publish_time = publish_time;
	}

	/*public Timestamp getLastUpdate() {
		return this.last_update;
	}

	public void setLastUpdate(Timestamp lastUpdate) {
		this.last_update = lastUpdate;
	}*/
	
	public Date getLast_update() {
		return last_update;
	}

	public void setLast_update(Date last_update) {
		this.last_update = last_update;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

}

File: keyansys/WebRoot/app/model/staticData/Paper.js
/*function Todate(v) {
　　　　　　　　if(v == null)
　　　　　　　　{
　　　　　　　　　　return null;
　　　　　　　　}

        var str = v.toString().substring(0,10);//因为我只要日期，所以我只取T之前的数据2013-04-01
        return new Date(Date.parse(str.replace(/-/g,"/")));//最后返回一个日期型的数据
    　　};
*/
Ext.define('keyansys.model.staticData.Paper', {
    extend: 'keyansys.model.staticData.Base',

    requires: [
        'Ext.data.validator.Length'
    ],

  //  entityName:'Paper',
    idProperty: 'paper_id',
   
    fields: [
   
        { name : 'paper_id'} ,
        { name: 'name'},
        { name: 'username'},
        { name: 'department'},
        { name: 'paper_name'},
        { name: 'first_author',defaultValue:"无"},
        { name: 'periodicalName'},
        { name: 'issn'},
        { name:'cn',defaultValue:"无"},
        { name: 'volume', type:"int"},
        { name:'publish_time',type:'date',dateFormat: 'Y-m-d',defaultValue: new Date()},
        { name: 'core',defaultValue:"校内刊物"},
       // { name: 'pursuant'},
        { name: 'approval',defaultValue:"未审核"}
     ],
     validators: {
        paper_name: [
            { type: 'presence', message: '该项数据不能为空。'},
            { type: 'length', min: 1}
        ],
        periodicalName: [
            { type: 'presence', message: '该项数据不能为空'},
            { type: 'length', min: 1}
        ],
        issn: [
            { type: 'presence', message: '该项数据不能为空'},
            { type: 'length', min: 1,max:20}
        ],
        volume: [
            { type: 'presence', message: '该项数据不能为空'},
            { type: 'length', min: 1}
        ],
        publish_time: [
           { type: 'presence', message: '该项数据不能为空'},
           { type: 'length', min: 1}
        ],
        core: [
                { type: 'presence', message: '该项数据不能为空'},
                { type: 'length', min: 1}
             ]
    }
});